scripts/preprocess_ef.py

```diff
@@ -0,0 +1,148 @@
+    )
+
+    for sample in samples:
+        traj_logs = open(sample, "r").read().splitlines()
+        xyz_idx = os.path.splitext(os.path.basename(sample))[0]
+        traj_path = os.path.join(args.data_path, f"{xyz_idx}.extxyz")
+        traj_frames = ase.io.read(traj_path, ":")
+
+        for i, frame in enumerate(traj_frames):
+            frame_log = traj_logs[i].split(",")
+            data_object = a2g.convert(frame)
+            # add atom tags
+            data_object.tags = torch.LongTensor(frame.get_tags())
+            # subtract off reference energy
+            if args.ref_energy:
+                ref_energy = float(frame_log[2])
+                data_object.y -= ref_energy
+
+            txn = db.begin(write=True)
+            txn.put(
+                f"{idx}".encode("ascii"),
+                pickle.dumps(data_object, protocol=-1),
+            )
+            txn.commit()
+            idx += 1
+            sampled_ids.append(",".join(frame_log[:2]) + "\n")
+
+    # Save count of objects in lmdb.
+    txn = db.begin(write=True)
+    txn.put("length".encode("ascii"), pickle.dumps(idx, protocol=-1))
+    txn.commit()
+
+    db.sync()
+    db.close()
+
+    return sampled_ids, idx
+
+
+def chunk_list(lst, num_splits):
+    n = max(1, len(lst) // num_splits)
+    return [lst[i : i + n] for i in range(0, len(lst), n)]
+
+
+if __name__ == "__main__":
+    # Parse a few arguments.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--data-path",
+        required=True,
+        help="Path to dir containing *.extxyz and *.txt files",
+    )
+    parser.add_argument(
+        "--out-path",
+        required=True,
+        help="Directory to save extracted features. Will create if doesn't exist",
+    )
+    parser.add_argument(
+        "--get-edges",
+        action="store_true",
+        help="Store edge indices in LMDB, ~10x storage requirement. Default: compute edge indices on-the-fly.",
+    )
+    parser.add_argument(
+        "--num-workers",
+        type=int,
+        default=1,
+        help="No. of feature-extracting processes or no. of dataset chunks",
+    )
+    parser.add_argument(
+        "--ref-energy", action="store_true", help="Subtract reference energies"
+    )
+    args = parser.parse_args()
+
+    xyz_logs = glob.glob(os.path.join(args.data_path, "*.txt"))
+
+    # Initialize feature extractor.
+    a2g = AtomsToGraphs(
+        max_neigh=50,
+        radius=6,
+        r_energy=True,
+        r_forces=True,
+        r_distances=False,
+        r_fixed=True,
+        r_edges=args.get_edges,
+    )
+
+    # Create output directory if it doesn't exist.
+    os.makedirs(os.path.join(args.out_path), exist_ok=True)
+
+    # Initialize lmdb paths
+    db_paths = [
+        os.path.join(args.out_path, "data.%04d.lmdb" % i)
+        for i in range(args.num_workers)
+    ]
+
+    # Chunk the trajectories into args.num_workers splits
+    chunked_traj_files = chunk_list(xyz_logs, args.num_workers)
```

Comment:
nit: consider changing variable name. Looks like you are using .txt files, so this should be `chunked_txt_files` ?